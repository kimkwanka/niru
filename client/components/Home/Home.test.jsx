import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ConnectedHome, { Home } from './Home';

const user = {
  name: 'TestUsername',
  authenticated: true,
};

const dispatch = () => {};

describe('<Home /> - Snapshot', () => {
  it('matches its snapshot', () => {
    const wrapper = shallow(<Home {...{ user, dispatch }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('<Home /> - Shallow render undecorated component', () => {
  it('renders without exploding', () => {
    const wrapper = shallow(<Home {...{ user, dispatch }} />);
    expect(wrapper).toHaveLength(1);
  });
  it('renders the user name', () => {
    const wrapper = shallow(<Home {...{ user, dispatch }} />);
    expect(wrapper.find('h4 span').at(0).text()).toBe(' TestUsername');
  });
  it('renders the user authentication status: true', () => {
    const wrapper = shallow(<Home {...{ user, dispatch }} />);
    expect(wrapper.find('h4 span').at(1).text()).toBe(' true');
  });
  it('renders the user authentication status: false', () => {
    const falseUser = {
      name: 'TestUsername',
      authenticated: false,
    };
    const wrapper = shallow(<Home user={falseUser} dispatch={dispatch} />);
    expect(wrapper.find('h4 span').at(1).text()).toBe(' false');
  });
});

describe('<Home /> - Mount Redux connected component', () => {
  const mockStore = configureStore();
  const initialState = { user };

  let store = null;
  let wrapper = null;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <ConnectedHome />
      </Provider>,
    );
  });

  it('renders without exploding', () => {
    expect(wrapper.find(Home)).toHaveLength(1);
  });

  it('handles input change', () => {
    wrapper.find('input').simulate('change', {
      target: { value: 'NewUsername' },
    });
    const actions = store.getActions();
    const expectedAction = {
      type: 'RENAME_USER',
      name: 'NewUsername',
    };
    expect(actions).toEqual([expectedAction]);
  });

  it('handles clicks', () => {
    wrapper.find('button').simulate('click');
    const actions = store.getActions();
    const expectedAction = {
      type: 'TOGGLE_AUTHENTICATED',
    };
    expect(actions).toEqual([expectedAction]);
  });
});
