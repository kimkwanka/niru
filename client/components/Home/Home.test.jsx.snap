// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> - Snapshot matches its snapshot 1`] = `
<div
  className="container margin-top-small"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Home
    </title>
  </HelmetWrapper>
  <h1>
    Home
  </h1>
  <h3
    className="center"
  >
    Data from (prehydrated) Redux store:
  </h3>
  <h4
    className="center normal"
  >
    user.name:
    <span
      className="bold"
    >
       TestUsername
    </span>
  </h4>
  <h4
    className="center normal"
  >
    user.authenticated:
    <span
      className="bold"
    >
       true
    </span>
  </h4>
  <div
    className="flex-column items-center margin-top-big"
  >
    <input
      defaultValue="TestUsername"
      onChange={[Function]}
      type="text"
    />
    <button
      className="margin-top-small button--accent"
      onClick={[Function]}
      type="button"
    >
      Toggle Authenticated
    </button>
  </div>
</div>
`;
